tools/common/StaticRegistrar.cs

```diff
@@ -2215,8 +2215,37 @@ void CheckNamespace (string ns, List<Exception> exceptions)
 				goto default;
 #if !NET
 			case "Chip":
-				h = "<CHIP/CHIP.h>";
-				break;
+				switch (App.Platform) {
+				case ApplePlatform.iOS:
+				case ApplePlatform.TVOS:
+					if (App.SdkVersion.Major <= 15)
+						h = "<CHIP/CHIP.h>";
+					else
+						goto default;
+					break;
+				case ApplePlatform.MacOSX:
+					if (App.SdkVersion.Major <= 12)
+						h = "<CHIP/CHIP.h>";
+					else
+						goto default;
+					break;
+				case ApplePlatform.WatchOS:
+					if (App.SdkVersion.Major <= 8)
+						h = "<CHIP/CHIP.h>";
+					else
+						goto default;
+					break;
```

Comment:
goto are dangerous are are perse not needed, you can write it in a more elegant way using the new switch statements. 
```suggestion
				switch (App.Platform) {
				case ApplePlatform.iOS when App.SdkVersion.Major <= 15:
				case ApplePlatform.TVOS when App.SdkVersion.Major <= 15:
					h = "<CHIP/CHIP.h>";
					break;
				case ApplePlatform.MacOSX when App.SdkVersion.Major <= 12:
					h = "<CHIP/CHIP.h>";
					break;
				case ApplePlatform.WatchOS when App.SdkVersion.Major <= 8):
					h = "<CHIP/CHIP.h>";
					break;
```
Then use the default as you have it.